pkg/maps/filter/sock_filter.go

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package filter
+
+import (
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/types"
+
+	"github.com/cilium/ebpf"
+
+	"github.com/cilium/hive/cell"
+)
+
+const (
+	// SockFilterMap is the BPF map name.
+	SockTermFilterMapName = "cilium_sock_term_filter"
+
+	// SockTermFilterMapSize is the maximum number of entries in the BPF map.
+	SockTermFilterMapSize = 1
+)
+
+var (
+	// SockTermFilter is the socket termination filter BPF map.
+	SockTermFilter *SockTermFilterMap
+
+	key = index(0)
+)
+
+// Cell provides the SockTermFilterMap which allows users to set the filter
+// used by the socket termination BPF program.
+var Cell = cell.Module(
+	"sock-term-filter",
+	"eBPF map containing the filter for socket termination",
+
+	cell.Provide(func(lifecycle cell.Lifecycle) *SockTermFilterMap {
```

Comment:
How does this deal with ordering with regards to loading the iterator program? 

It might be worth considering not adding this map package at all, but instead returning the map via `LoadSockTerm`, e.g. grabbing it from the collection instead. This way there won't be ordering issues.

Reply:
> It might be worth considering not adding this map package at all, but instead returning the map via LoadSockTerm

Yeah, this seems cleaner and lets me git rid of the mutex and synchronization cruft. I will have to think a bit about how this impacts the test code, but I think it should work overall.

Reply:
I ended up reworking this so that `LoadSockTerm()` returns a map and squashing the commit that adds the `socktermfilter` package into the commit that introduces `BPFSocketDestroyer` and co, since there wasn't much left in that package except type definitions after the changes.